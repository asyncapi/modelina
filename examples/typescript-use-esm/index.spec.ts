const spy = jest.spyOn(global.console, 'log').mockImplementation(() => {
  return;
});
import { generate } from './index';

describe('Should be able to render models to ESM module system', () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('and should log expected output to console', async () => {
    await generate();
    expect(spy.mock.calls.length).toEqual(2);
    expect(spy.mock.calls[0]).toMatchSnapshot();
    expect(spy.mock.calls[1]).toMatchSnapshot();
  });
});
