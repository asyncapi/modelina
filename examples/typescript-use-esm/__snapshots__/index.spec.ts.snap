// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to render models to ESM module system and should log expected output to console 1`] = `
Array [
  "import Person from './Person';
class Root {
  private _person?: Person;

  constructor(input: {
    person?: Person,
  }) {
    this._person = input.person;
  }

  get person(): Person | undefined { return this._person; }
  set person(person: Person | undefined) { this._person = person; }
}
export default Root;
",
]
`;

exports[`Should be able to render models to ESM module system and should log expected output to console 2`] = `
Array [
  "
class Person {
  private _email?: string;

  constructor(input: {
    email?: string,
  }) {
    this._email = input.email;
  }

  get email(): string | undefined { return this._email; }
  set email(email: string | undefined) { this._email = email; }
}
export default Person;
",
]
`;
